docs/architecture/how/cross-shard.md

```diff
@@ -0,0 +1,301 @@
+
+```
+Chunk: Ok(
+    V2(
+        ShardChunkV2 {
+            chunk_hash: ChunkHash(
+                8mgtzxNxPeEKfvDcNdFisVq8TdeqpCcwfPMVk219zRfV,
+            ),
+            header: V3(
+                ShardChunkHeaderV3 {
+                    inner: V2(
+                        ShardChunkHeaderInnerV2 {
+                            prev_block_hash: CgTJ7FFwmawjffrMNsJ5XhvoxRtQPXdrtAjrQjG91gkQ,
+                            prev_state_root: 99pXnYjQbKE7bEf277urcxzG3TaN79t2NgFJXU5NQVHv,
+                            outcome_root: 11111111111111111111111111111111,
+                            encoded_merkle_root: 67zdyWTvN7kB61EgTqecaNgU5MzJaCiRnstynerRbmct,
+                            encoded_length: 187,
+                            height_created: 1676,
+                            shard_id: 0,
+                            gas_used: 0,
+                            gas_limit: 1000000000000000,
+                            balance_burnt: 0,
+                            outgoing_receipts_root: 8s41rye686T2ronWmFE38ji19vgeb6uPxjYMPt8y8pSV,
+                            tx_root: HyS6YfQbfBRniVSbWRnxsxEZi9FtLqHwyzNivrF6aNAM,
+                            validator_proposals: [],
+                        },
+                    ),
+                    height_included: 0,
+                    signature: ed25519:uUvmvDV2cRVf1XW93wxDU8zkYqeKRmjpat4UUrHesJ81mmr27X43gFvFuoiJHWXz47czgX68eyBN38ejwL1qQTD,
+                    hash: ChunkHash(
+                        8mgtzxNxPeEKfvDcNdFisVq8TdeqpCcwfPMVk219zRfV,
+                    ),
+                },
+            ),
+            transactions: [
+                SignedTransaction {
+                    transaction: Transaction {
+                        signer_id: AccountId(
+                            "shard0",
+                        ),
+                        public_key: ed25519:Ht8EqXGUnY8B8x7YvARE1LRMEpragRinqA6wy5xSyfj5,
+                        nonce: 11,
+                        receiver_id: AccountId(
+                            "shard1",
+                        ),
+                        block_hash: 6d5L1Vru2c4Cwzmbskm23WoUP4PKFxBHSP9AKNHbfwps,
+                        actions: [
+                            Transfer(
+                                TransferAction {
+                                    deposit: 500000000000000000000000000,
+                                },
+                            ),
+                        ],
+                    },
+                    signature: ed25519:63ssFeMyS2N1khzNFyDqiwSELFaUqMFtAkRwwwUgrPbd1DU5tYKxz9YL2sg1NiSjaA71aG8xSB7aLy5VdwgpvfjR,
+                    hash: 6NSJFsTTEQB4EKNKoCmvB1nLuQy4wgSKD51rfXhmgjLm,
+                    size: 114,
+                },
+            ],
+            receipts: [],
+        },
+    ),
+)
+```
+
+```
+Side note: When we're changing the transaction into receipt, we also use this moment to deduct prepaid gas fees and transfered tokens from the 'signer' account.
```

Comment:
> and transfered tokens

This deserves more than a side note.

Reply:
yep - we'll create a separate article about how much gas we charge etc.